analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyNullableValueAccess.cs

```diff
@@ -425,6 +419,113 @@ class TernaryOperator
+
+        static int? Unknown() => null;
+    }
+
+    void WithFuncOfNullable(Func<int?> f)
+    {
+        f = Unknown();
+        _ = f().Value;                // Compliant, f unknown
+        f = Unknown();
+        _ = f.Invoke().Value;         // Compliant, f unknown
+        f = Unknown();
+        _ = f?.Invoke().Value;        // Compliant, Value never called when f is null
+        f = Unknown();
+        _ = (f?.Invoke()).Value;      // Noncompliant, ?. branches the state of f, and Value called on a null path
+        f = Unknown();
+        _ = (f?.Invoke()).ToString(); // Compliant, ?. branches the state of f, but Value never called anyway
+        f = Unknown();
+        _ = f?.Invoke()?.ToString();  // Compliant, ?. branches the state of f, but Value never called anyway
+
+        static Func<int?> Unknown() => null;
+    }
+
+    void WithFuncOfFuncOfNullable(Func<Func<int?>> f)
+    {
+        f = Unknown();
+        _ = f()().Value;               // Compliant, f unknown
+        f = Unknown();
+        _ = f?.Invoke()().Value;       // Compliant, Value never called when f is null
+        f = Unknown();
+        _ = (f?.Invoke()()).Value;     // Noncompliant, ?. branches the state of f
+        f = Unknown();
+        _ = (f?.Invoke())().Value;     // Compliant, ?. branches the state of f, but f.Invoke() is unknown
+
+        static Func<Func<int?>> Unknown() => null;
+    }
+}
+
+class ValueTuple
+{
+    void MembersAccess((int? first, int? second)? tuple, int? i)
+    {
+        tuple = Unknown();
+        _ = tuple.Value;              // Compliant, unknown
+        tuple = Unknown();
+        _ = tuple.Value.first.Value;  // Compliant, tuple unknown, then first unknown
+        _ = tuple.Value.second.Value; // Compliant, tuple unknown, then second unknown
+        tuple = (null, i);
+        _ = tuple.Value;              // Compliant, non-empty
+        _ = tuple.Value.first.Value;  // FN, empty
+        _ = tuple.Value.second.Value; // Compliant, second still unknown
+        tuple = (i, null);
+        _ = tuple.Value;              // Compliant, non-empty
+        _ = tuple.Value.first.Value;  // Compliant, first still unknown
+        _ = tuple.Value.second.Value; // FN, empty
+
+        static (int? first, int? second)? Unknown() => null;
+    }
+}
+
+class NullCoaleascingAssignment
+{
+    void WithNullableOfInt(int? i1, int? i2, int? i3)
+    {
+        i1 ??= 42;
+        _ = i1.Value;  // Compliant, always non-empty
+        i1 = null;
+        i1 ??= null;
+        _ = i1.Value;  // Noncompliant, always empty
+        i1 = 42;
+        i1 ??= null;
+        _ = i1.Value;  // Compliant, always non-empty
+
+        i2 ??= i3;
+        _ = i2.Value;  // Compliant, both i2 and i3 are unknown
+    }
+
+    void WithNullableOfValueTuple()
+    {
+        (int? first, int? second)? tuple = null;
+        tuple ??= (42, 42);
+        _ = tuple.Value.first.Value;  // Noncompliant, FP: non-empty
```

Comment:
To fix this FP, we need to build `NotNull` on every tuple operation. 

Do I understand it correctly?

Reply:
Ok, thanks. Once step 1 is implemented, step 2 should just work